### PR TITLE
Better upgrade cli command with progress and other stuff...

### PR DESCRIPTION
**Pull request by zohaibakber**

### CODE REVIEW

File: cmd/sst/ui/ui.go [hunk 1]
```diff
@@ -24,8 +24,9 @@ const (
 )
 
 const (
-	IconX     = "×"
+	IconX     = "x"
```
Comment (jayair) [inline cmd/sst/ui/ui.go]: lol we spent too much time picking this `x`, how come you wanna change it?

Comment (zohaibakber) [reply cmd/sst/ui/ui.go]: @jayair it was doing naughty stuff in my terminal, sorry for hurting your feelings :p 
Also why tf are you people spending time on a cross icon decision?

Comment (jayair) [reply cmd/sst/ui/ui.go]: Oh really? Can I see a screenshot? I'm curious what it's doing.

Comment (zohaibakber) [reply cmd/sst/ui/ui.go]: image

Comment (zohaibakber) [reply cmd/sst/ui/ui.go]: maybe its my term sometimes it works sometimes it meses it up
but whatever I will revert it

File: pkg/global/upgrade.go [hunk 2]
```diff
@@ -100,8 +116,21 @@ func Upgrade(version string) (string, error) {
 	return version, nil
 }
 
-func untar(reader io.Reader, target string) error {
-	tarReader := tar.NewReader(reader)
+func untar(reader io.Reader, target string, totalSize int64) error {
+	p := mpb.New(mpb.WithWidth(64))
+	bar := p.AddBar(totalSize,
```
Comment (thdxr) [inline pkg/global/upgrade.go]: we do not want this function printing anything out to the terminal - everything in `pkg` should be agnostic of where it's being called from

to support a progress thing like this the `Upgrade` command would have to take a callback or something

File: pkg/project/create.go [hunk 3]
```diff
@@ -79,7 +80,11 @@ func Create(templateName string, home string) error {
 				return err
 			}
 			slog.Info("patching", "file", patchStep.File)
-			data, err := os.ReadFile(patchStep.File)
+			b, err := os.ReadFile(patchStep.File)
+			if err != nil {
+				return err
+			}
+			data, err := hujson.Standardize(b)
```
Comment (thdxr) [inline pkg/project/create.go]: can you send this in another PR

Comment (zohaibakber) [reply pkg/project/create.go:87]: @thdxr https://github.com/sst/ion/pull/88

File: pkg/global/upgrade.go (hunk 2 above)
Comment (zohaibakber) [reply pkg/global/upgrade.go]: @thdxr I guess https://github.com/sst/ion/commit/9ae9d3d1c972cdd4318b9f8d54e2c15814f4edcf after this pr can closed?
